### PR TITLE
Add assertion that checks that the same shard with same id is not added to same node

### PR DESCRIPTION
**Pull request by ywelsch**
Adds an assertion that checks that the same shard with same id is not added to same node. Previously we would just silently ignore the second shard being added.

### CODE REVIEW

File: core/src/main/java/org/elasticsearch/cluster/routing/IndexShardRoutingTable.java [hunk 1]
```diff
@@ -590,9 +583,28 @@ public Builder removeShard(ShardRouting shardEntry) {
         }
 
         public IndexShardRoutingTable build() {
+            // don't allow more than one shard copy with same id to be allocated to same node
+            assert distinctNodes() : "more than one shard with same id assigned to same node (shards: " + shards + ")";
             return new IndexShardRoutingTable(shardId, Collections.unmodifiableList(new ArrayList<>(shards)));
         }
 
+        private boolean distinctNodes() {
```
Comment (dakrone) [inline core/src/main/java/org/elasticsearch/cluster/routing/IndexShardRoutingTable.java]: If you passed in `shards` here, you could make this static and public, and then unit test it. What do you think?

**Review by dakrone — APPROVED** [review]
LGTM, left one (optional) comment

File: core/src/main/java/org/elasticsearch/cluster/routing/IndexShardRoutingTable.java (hunk 1 above)
Comment (ywelsch) [inline core/src/main/java/org/elasticsearch/cluster/routing/IndexShardRoutingTable.java]: good idea. I've added this.

**Comment by ywelsch** [comment]
Thanks @dakrone

**Pull request merged**